src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -236,6 +239,19 @@ private bool TryNavigateToLocation(Workspace workspace, DocumentId documentId, F
 
             using (OpenNewDocumentStateScope(options ?? workspace.Options))
             {
+                if (workspace.CurrentSolution.GetDocument(documentId) == null)
+                {
+                    var generatedDocument = workspace.CurrentSolution.GetProject(documentId.ProjectId)
+                                                                     .GetSourceGeneratedDocumentAsync(documentId, CancellationToken.None)
```

Comment:
being uncancellable is so concerning.  we may have to drive a discussion on navigation to things that are very expensive.

Reply:
Actually I'll update the comment here -- presumably this is cheap, because if we're navigating to the file that means we've already computed the contents of it -- otherwise how do we know to go there in the first place?